src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Queue.cs

```diff
@@ -462,7 +462,7 @@ public bool MoveNext()
                 // _index represents the 0-based index into the queue, however the queue
                 // doesn't have to start from 0 and it may not even be stored contiguously in memory.
 
-                int arrayIndex = _q._head + _index; // this is the actual index into the queue's backing array
+                long arrayIndex = (long)_q._head + _index; // this is the actual index into the queue's backing array
```

Comment:
This will penalize 32-bit platforms. Can `uint` be used instead?

Reply:
Thanks, yes we don't need to widen the storage to fix the overflow.